include/miroil/miroil/mirbuffer.h

```diff
@@ -21,41 +21,38 @@
 #include <mir/version.h>
 #include <memory>
 
-#include <GL/gl.h>
-
-namespace mir { namespace graphics { class Buffer; }}
+namespace mir {
+    namespace graphics {
+        class Buffer;
+
+        namespace gl {
+            class Texture;
+        }
+    }
+}
 
 namespace miroil
 {
 class GLBuffer
 {
 public:
-    GLBuffer();
     ~GLBuffer();
     explicit GLBuffer(std::shared_ptr<mir::graphics::Buffer> const& buffer);
 
-    operator bool() const;
     bool has_alpha_channel() const;
     mir::geometry::Size size() const;
+    void bind();
 
-    void reset();
     void reset(std::shared_ptr<mir::graphics::Buffer> const& buffer);
-    void bind();
-#if MIR_SERVER_VERSION < MIR_VERSION_NUMBER(2, 3, 0)
-    void gl_bind_tex();
-#endif
 
-private:
-    void init();
-    void destroy();
+    void reset();
+    bool empty();
+
+    static std::shared_ptr<GLBuffer> from_mir_buffer(std::shared_ptr<mir::graphics::Buffer> const& buffer);
 
+protected:
     std::shared_ptr<mir::graphics::Buffer> wrapped;
-    GLuint m_textureId;
-#if MIR_SERVER_VERSION < MIR_VERSION_NUMBER(2, 3, 0)
-    bool m_isOldTex = false;
-    bool m_inited = false;
-#endif
 };
 }
 
-#endif //MIROIL_GLBUFFER_H
+#endif //MIRAL_GLBUFFER_H
```

Comment:
Comment should match this seen at the top of the file
```c
#ifndef MIROIL_GLBUFFER_H
#define MIROIL_GLBUFFER_H
```

Reply:
done